viewer/packages/360-images/src/Image360Facade.ts

```diff
@@ -50,23 +50,28 @@ export class Image360Facade<T> {
 
   public async delete(entity: Image360Entity): Promise<void> {
     await this._image360Cache.purge(entity);
-    const collectionContainingEntity = this._image360Collections.filter(collection =>
-      collection.image360Entities.includes(entity)
-    );
-    collectionContainingEntity.forEach(collection => {
-      collection.remove(entity);
-    });
-    const disposeableCollections = collectionContainingEntity.filter(
-      collection => collection.image360Entities.length === 0
-    );
-    disposeableCollections.forEach(collection => collection.dispose());
-    pullAll(this._image360Collections, disposeableCollections);
+    const collection = this.getCollectionContainingEntity(entity);
+    collection.remove(entity);
+    if (collection.image360Entities.length === 0) {
+      collection.dispose();
+      pull(this._image360Collections, collection);
+    }
   }
 
   public preload(entity: Image360Entity, lockDownload?: boolean): Promise<void> {
     return this._image360Cache.cachedPreload(entity, lockDownload);
   }
 
+  public getCollectionContainingEntity(entity: Image360Entity): DefaultImage360Collection {
+    const imageCollection = this._image360Collections.filter(collection =>
+      collection.image360Entities.includes(entity)
+    );
+    if (imageCollection.length !== 1) {
+      throw new Error(`360Image entity contained in invalid number of collections: {${imageCollection.length}}`);
```

Comment:
```suggestion
      throw new Error(`360Image entity contained an invalid number of collections: {${imageCollection.length}}`);
```

Reply:
Changed to: "Failed to get Collection for Image360Entity. The entity is present in ${imageCollection.length} collections."